src/modules/twap/updaters/PartOrdersUpdater.tsx

```diff
@@ -0,0 +1,119 @@
+import { useAtomValue, useUpdateAtom } from 'jotai/utils'
+import { useEffect } from 'react'
+
+import { Order } from '@cowprotocol/contracts'
+import { OrderParameters, SupportedChainId } from '@cowprotocol/cow-sdk'
+
+import { isTruthy } from 'legacy/utils/misc'
+
+import { useWalletInfo } from 'modules/wallet'
+
+import { computeOrderUid } from 'utils/orderUtils/computeOrderUid'
+
+import { twapOrdersListAtom } from '../state/twapOrdersListAtom'
+import { TwapPartOrderItem, twapPartOrdersAtom } from '../state/twapPartOrdersAtom'
+import { TwapOrderItem } from '../types'
+
+export function PartOrdersUpdater() {
+  const { chainId, account } = useWalletInfo()
+  const twapOrdersList = useAtomValue(twapOrdersListAtom)
+  const updateTwapPartOrders = useUpdateAtom(twapPartOrdersAtom)
+
+  useEffect(() => {
+    if (!chainId || !account) return
+
+    const twapOrders = Object.values(twapOrdersList)
+
+    const ordersParts$ = twapOrders.map((twapOrder) => {
+      return generateTwapOrderParts(twapOrder, account.toLowerCase(), chainId)
```

Comment:
shouldn't you lowercaase inside the mehtod instead?

Reply:
We do a lowercase once in `PartOrdersUpdater`. In terms of performance, it should be better.

Reply:
In terms of performance, it's better to call `toLowerCase()` once. I would like to move the result of `account.toLowerCase()` in const outside of `map()`. Wdyt?